lib/wallet/src/Cardano/Wallet/DB/Store/DeltaUTxO/Store.hs

```diff
@@ -0,0 +1,328 @@
+write
+    wid
+    ( UTxOHistory
+            { history
+            , creationSlots
+            , spentSlots
+            , tip
+            , finality
+            , boot
+            }
+        ) = do
+        deleteWhere [DeltaUTxOSlotsWallet ==. wid]
+        insert_ $ DeltaUTxOSlots wid finality tip
+        deleteWhere [DeltaUTxOValueWalletId ==. wid]
+        insertMany_ $ do
+            (txIn@TxIn {inputId, inputIx}, txOut) <-
+                Map.assocs $
+                    unUTxO $
+                        history <> boot
+            creation <-
+                maybeToList $
+                    Map.lookup txIn $
+                        reverseMapOfSets creationSlots
+            let
+                spent = Map.lookup txIn $ reverseMapOfSets spentSlots
+            pure $
+                DeltaUTxOValue
+                    wid
+                    creation
+                    (maybe Unspent Spent spent)
+                    (TxId inputId)
+                    inputIx
+                    (encodeTxOut txOut)
+
+encodeTxOut = view strict . serializeTxOut
+encodeTxOut :: TxOut -> ByteString
+
+update :: WalletId -> Maybe UTxOHistory -> DeltaUTxOHistory -> SqlPersistT IO ()
+update wid = updateLoad (load wid) throwIO updateJust
+  where
+    updateJust :: UTxOHistory -> DeltaUTxOHistory -> SqlPersistT IO ()
+    updateJust old =
+        \case
+            AppendBlock newTip delta ->
+                constrainingAppendBlock (pure ()) old newTip $ do
+                    let
+                        reiceved = received delta
+                        spent = excluded delta
+                    insertMany_ $ do
+                        (TxIn {inputId, inputIx}, txOut) <-
+                            Map.assocs $
+                                unUTxO reiceved
+                        pure $
+                            DeltaUTxOValue
+                                wid
+                                (At newTip)
+                                Unspent
+                                (TxId inputId)
+                                inputIx
+                                (encodeTxOut txOut)
+                    {- TODO: Prove this is worse than the below.
+                    forM_ spent $ \(TxIn txId txIx) ->
+                        Sql.update
+                            (DeltaUTxOValueKey wid (TxId txId) txIx)
+                            [Update DeltaUTxOValueSpent (Just slot) Assign]
+                    -}
+                    let
+                        (txIds, txIxs) =
+                            unzip $
+                                (\(TxIn txId txIx) -> (txId, txIx))
+                                    <$> Set.toList spent
+                    updateWhere
+                        [ Filter DeltaUTxOValueWalletId (FilterValue wid) Eq
+                        , Filter
+                            DeltaUTxOValueTxInTx
+                            (FilterValues $ TxId <$> txIds)
+                            In
```

Comment:
```suggestion
                        , DeltaUTxOValueTxInTx <-. (TxId <$> txIds)
```

… and also for `In`. It's not perfect, but I think that it's better than `In` at least.